src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const partsDate = date.split(fromFormat[3]);
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i <= 3; i++) {
+    if (fromFormat[i] === 'DD') {
+      day = partsDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = partsDate[i];
+    }
+
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = partsDate[i];
+    }
+  }
+
+  const newPartsDate = [];
+
+  for (let i = 0; i <= 3; i++) {
```

Comment:
Similar to the previous loop, the condition `i <= 3` assumes a fixed length for the `toFormat` array. Use `toFormat.length` to ensure the loop adapts to different format lengths.